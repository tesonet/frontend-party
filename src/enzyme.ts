import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import fetchMock from "jest-fetch-mock";

declare let global: { fetch: {} };

global.fetch = fetchMock;
configure({ adapter: new Adapter() });
