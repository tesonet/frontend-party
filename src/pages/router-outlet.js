import React from "react";
import {connect} from "react-redux";
import {Redirect, Router} from "@reach/router";
import styled from "styled-components";

import LoginPage from "./login-page";
import ServerListPage from "./server-list-page";


const StyledRouter = styled(Router)`
   height: 100%;
  `;

const LoginRoute = ({logged_in}) => (
    logged_in === false
        ? <LoginPage path={"/"}/>
        : <Redirect noThrow to='/dashboard'/>
);

const GuardedRoute = ({component: Component, logged_in, path}) => (
    logged_in === true
        ? <Component path={path}/>
        : <Redirect noThrow to='/'/>
);

export const RouterOutlet = ({logged_in}) => {
    return (
        <StyledRouter>
            <LoginRoute logged_in={logged_in} path={"/"}/>
            <GuardedRoute logged_in={logged_in} path={"/dashboard"} component={ServerListPage}/>
        </StyledRouter>
    );
};

export default connect(
    state => state.auth,
    null
)(RouterOutlet);
