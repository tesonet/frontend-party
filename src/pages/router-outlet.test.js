import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";

import {RouterOutlet} from "./router-outlet";


const component = shallow(<RouterOutlet/>);

it('renders correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
});
