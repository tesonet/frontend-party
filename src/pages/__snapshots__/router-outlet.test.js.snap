// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Styled(Router)>
  <LoginRoute
    path="/"
  />
  <GuardedRoute
    component={[Function]}
    path="/dashboard"
  />
</Styled(Router)>
`;
